ftd/src/interpreter/test.rs

```diff
@@ -258,3 +258,31 @@ fn evalexpr_test() {
         }
             */
 }
+
+#[test]
+fn test_extract_kwargs() {
+    let doc = interpret_helper(
+        "foo",
+        r#"-- component foo:
+    kw-args data:
+
+    -- ftd.text: Hello world
+
+    -- end: foo
+
+    -- foo:
+    bar: Hello
+    baz: World
+"#,
+    )
+    .unwrap();
+
+    let tdoc = doc.tdoc();
+    let instructions = doc.get_instructions("foo#foo");
+    let instruction = instructions.first().unwrap();
```

Comment:
This is still not good. tdoc is temporary doc, and is internal property. Can you write a new crate, that uses ftd crate a dependency, and see what code you will write. Test against that code.